### ISSUE TITLE
Crash when connecting to nodownload.twcclan.org

### ISSUE THREAD

**Issue by KiralyCraft** [etlegacy/etlegacy#1665]
I'm currently using commit 6ffb0c998ac1465ddbc9e96977fca35d0727f569 on Manjaro as well as Windows 10. Joining the same server on both, `nodownload.twcclan.org`, works wonderfully on Manjaro, whereas on Windows it crashes to the desktop with no errors once joining the server. 

On Windows, the issue persists from version 2.77 all the way to that commit, which at the time of writing, is the latest. I tried 2.76, where I get greeted by a message about invalid characters: 
![image](https://user-images.githubusercontent.com/10259742/120896180-18628280-c629-11eb-9b27-d76e6badcd6f.png)

On version 2.75, the game still crashes, unless I manually copy the `etmain` folder created by version 6ffb0c998ac1465ddbc9e96977fca35d0727f569 on top of the one existing inside the "portable" (archived) version of 2.75. Then it works. This is a version I've put together with portable 2.75 and the etmain "trick": https://1fichier.com/?qf7bx1aetxcek9cf6evq

I managed to get a logfile from 6ffb0c998ac1465ddbc9e96977fca35d0727f569, it says:

```       0 logfile opened on Sat Jun 05 17:28:58 2021
       0 ]/logfile 2
       0 ]/logfile 2
       0 ]/logfile 2
       0 ]/connect nodownload.twcclan.org
       0 nodownload.twcclan.org resolved to 178.63.136.66:27960
       0 Connected to a pure server.
       0 Current working directory:
       0     C:\Program Files (x86)\ETLegacy
       0 Current search path:
       0     C:\Users\PC\Documents\ETLegacy\jaymod\~~~~~~~twc_menu.pk3 (54 files)
       0         on the pure list
       0     C:\Users\PC\Documents\ETLegacy\jaymod\z_twc_pumpkin_noclip.pk3 (5 files)
       0         on the pure list
       0     C:\Users\PC\Documents\ETLegacy\jaymod\z_et_mor2_night_final_voice.pk3 (16 files)
       0         on the pure list
       0     C:\Users\PC\Documents\ETLegacy\jaymod\jaymod-2.2.0_reds_v102.pk3 (253 files)
       0         on the pure list
       0     C:\Users\PC\Documents\ETLegacy\jaymod\enhmodv109.pk3 (13 files)
       0         on the pure list
       0     C:\Users\PC\Documents\ETLegacy\jaymod\admin_sounds.pk3 (6 files)
       0         on the pure list
       0     C:\Users\PC\Documents\ETLegacy\etmain\dlcache\baserace.pk3 (129 files)
       0         on the pure list
       0     .\etmain\pak2.pk3 (22 files)
       0         on the pure list
       0     .\etmain\pak1.pk3 (10 files)
       0         on the pure list
       0     .\etmain\pak0.pk3 (3725 files)
       0         on the pure list
       0     C:\Users\PC\Documents\ETLegacy\jaymod
       0     .\jaymod
       0     C:\Users\PC\Documents\ETLegacy\etmain\dlcache
       0     C:\Users\PC\Documents\ETLegacy\etmain
       0     .\etmain
       0 
       0 handle 2: etconsole.log
       0 4233 files in pk3 files
       0 --------------------------------
       0 Creating PID file 'profiles/Bacon/profile.pid'
       0 Com_TrackProfile: Deleting old pid file [legacy] [profiles/Bacon/profile.pid]
       0 Client download complete
       0 RE_Shutdown( 0 )
       0 Hunk_Clear: reset the hunk ok
       0 ----- Initializing Renderer ----
       0 GL_VENDOR: NVIDIA Corporation
       0 GL_RENDERER: NVIDIA GeForce GTX 1060 3GB/PCIe/SSE2
       0 GL_VERSION: 4.6.0 NVIDIA 466.27
       0 GL_SHADING_LANGUAGE_VERSION: 4.60 NVIDIA
       0 GL_MAX_TEXTURE_SIZE: 32768
       0 GL_MAX_ACTIVE_TEXTURES_ARB: 4
       0 PIXELFORMAT: color(24-bits) Z(24-bit) stencil(0-bits)
       0 MODE: -2, SCREEN: 1920 x 1080 windowed (ratio 1.7778) Hz:N/A
       0 GAMMA: hardware w/ 0 overbright bits
       0 texturemode: GL_LINEAR_MIPMAP_LINEAR
       0 picmip: 0
       0 texture bits: 32
       0 multitexture: enabled
       0 compiled vertex arrays: enabled
       0 texenv add: enabled
       0 compressed textures: enabled
       0 Initializing Shaders
       0 --------------------------------
       0 Sys_LoadGameDll -> FS_CL_ExtractFromPakFile(C:\Users\PC\Documents\ETLegacy, jaymod, ui_mp_x86.dll)
       0 Sys_LoadDll(C:\Users\PC\Documents\ETLegacy\jaymod\ui_mp_x86.dll)... succeeded
       0 Sys_LoadDll(jaymod/ui) found vmMain function at 720E4B00
       0 ^3WARNING: [S_AL_BufferUseDefault] Using default sound for sound/misc/menu2.wav
       0 ^1ERROR S_AL_BufferLoad: S_CodecLoad failed for "sound/misc/menu2.wav"
       0 Total favorite servers restored: 0
       0 Sys_LoadGameDll -> FS_CL_ExtractFromPakFile(C:\Users\PC\Documents\ETLegacy, jaymod, cgame_mp_x86.dll)
       0 Sys_LoadDll(C:\Users\PC\Documents\ETLegacy\jaymod\cgame_mp_x86.dll)... succeeded
       0 Sys_LoadDll(jaymod/cgame) found vmMain function at 1C2E33F0
       0 ----- Client Shutdown ----------
       0 Current working directory:
       0     C:\Program Files (x86)\ETLegacy
       0 Current search path:
       0     C:\Users\PC\Documents\ETLegacy\jaymod
       0     C:\Users\PC\Documents\ETLegacy\jaymod\~~~~~~~twc_menu.pk3 (54 files)
       0     C:\Users\PC\Documents\ETLegacy\jaymod\z_twc_pumpkin_noclip.pk3 (5 files)
       0     C:\Users\PC\Documents\ETLegacy\jaymod\z_et_mor2_night_final_voice.pk3 (16 files)
       0     C:\Users\PC\Documents\ETLegacy\jaymod\jaymod-2.2.0_reds_v102.pk3 (253 files)
       0     C:\Users\PC\Documents\ETLegacy\jaymod\enhmodv109.pk3 (13 files)
       0     C:\Users\PC\Documents\ETLegacy\jaymod\admin_sounds.pk3 (6 files)
       0     .\jaymod
       0     C:\Users\PC\Documents\ETLegacy\etmain
       0     .\etmain
       0     .\etmain\pak2.pk3 (22 files)
       0     .\etmain\pak1.pk3 (10 files)
       0     .\etmain\pak0.pk3 (3725 files)
       0 
       0 handle 2: etconsole.log
       0 4104 files in pk3 files
       0 --------------------------------
       0 Sys_LoadGameDll -> FS_CL_ExtractFromPakFile(C:\Users\PC\Documents\ETLegacy, jaymod, ui_mp_x86.dll)
       0 Sys_LoadDll(C:\Users\PC\Documents\ETLegacy\jaymod\ui_mp_x86.dll)... succeeded
       0 Sys_LoadDll(jaymod/ui) found vmMain function at 720E4B00
       0 Total favorite servers restored: 0
       0 RE_Shutdown( 1 )
       0 SDL input devices shut down.
       0 --------------------------------
       0 PID file removed.
       0 Network shutdown
```

To recap, the only way to join this particular server on Windows is to use 2.75 "portable" with 2.77.1's etmain folder, whereas on Linux it works right out of the box with the latest snapshot, 6ffb0c998ac1465ddbc9e96977fca35d0727f569

**Comment by rmarquis** [comment]
There is still a `Invalid file name detected` drop in `Sys_ListFiles` in both `sys_unix.c` and `sys_win32.c`.

Shouldn't this be completely unnecessary with the container system? It's been implemented to get rid of that very technically ugly workaround that also happened to enrage many players. See #1161.

**Comment by isRyven** [comment]
We have already eased the checks in https://github.com/etlegacy/etlegacy/commit/f571bb142b9ca646cfffc74e43964fe65fb273cb, so i am not entirely sure what's going on here.

**Comment by rmarquis** [comment]
Ah right, we still need to drop in case of some other naughty characters, many of which are non-printable.

The whole report can be ignored since it's a mix of older versions, and I guess there isn't enough info to understand why the initial crash happens (first two line of the report).
